domain/application/errors/errors.go

```diff
@@ -64,6 +64,10 @@ const (
 	// on does not exist.
 	UnitNotFound = errors.ConstError("unit not found")
 
+	// UnitNameNotValid describes an error that occurs when the unit name is
+	// not valid.
+	UnitNameNotValid = errors.ConstError("unit name not valid")
+
```

Comment:
question: Why add this error type now? There are approximately 40 other places that Validate is called with unit name which just return the error from Validate.

Feels like it's not needed, especially as it's not used by the caller.